lib/HLSL/DxilGenerationPass.cpp

```diff
@@ -106,8 +106,46 @@ class ResourcePromoter : public LoadAndStorePromoter {
     // Mark handle map.
     // If cannot find, will return false in run();
     if (Instruction *I = dyn_cast<Instruction>(V)) {
-      if (handleMap.count(I))
-        handleMap[NewInst] = handleMap[I];
+      if (handleMap.count(I)) {
+        Instruction *handle = cast<Instruction>(handleMap[I]);
+        // Clone the handle to save debug info of LI.
```

Comment:
LI was ambiguous at first. Spell it out as LoadInst (if that's what it stands for).